Libraries/Drei/drei-redisplay.lisp

```diff
@@ -29,6 +29,39 @@
 
 (in-package :drei)
 
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;  Wrapper around DRAW-TEXT* and TEXT-SIZE that
+;;;  handles font replacement
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defun iterate-font-replacement (stream string start end text-style fn)
+  (let* ((blocks (mcclim-font:find-replacement-text-styles stream (subseq string start end) :text-style text-style))
+         (text-style (or text-style (clim:medium-text-style stream)))
+         (size (clim:text-style-size text-style)))
+    (loop
+      with curr-x = 0
+      for (string family style) in blocks
+      for new-text-style = (if family (clim:make-text-style family style size) text-style)
+      do (progn
+           (funcall fn string curr-x new-text-style)
+           (multiple-value-bind (width)
```

Comment:
is m-v-b needed here if you are using only the primary return value?

Reply:
It's technically not needed, but I do it in order to make it clear that I only want the width and the height is not interesting. I also noticed that other code in McCLIM uses the same style so I figured it was OK.

I'll be happy to change it if you feel it's better that way.

Reply:
Fair point. I have fixed this in 5eeda5bfa47183c462346b469ec6a034d28aa269